uiowa/uiowa
# Create ThankYouForm block and configuration in sitenow_dispatch.

Resolves #4127 





# How to test
**Note** that the Dispatch communication is set to test mode and all emails will be sent to ITS web.

Make sure https://git.uiowa.edu/its-web/data.its.uiowa.edu/-/merge_requests/30 is deployed to the test server.

Add below config overrides to local.settings.php for site. Values can be obtained from uiowa.dev [secrets.settings.php file](https://docs.acquia.com/resource/secrets/#secrets-settings-php-file).
```
$config['sitenow_dispatch.settings']['thanks']['api_key'] = 'xxx';
$config['sitenow_dispatch.settings']['thanks']['hr_token'] = 'xxx';
```

### Stories
- As a webmaster configuring sitenow_dispatch, I cannot enable/disable the Thank You Form nor configure it.
- As a webmaster/editor/etc., I cannot place the ThankYouForm Dispatch block if it has not been enabled.
- As an administrator configuring sitenow_dispatch, I can enable/disable the Thank You Form and set its configuration.
- As a webmaster configuring sitenow_dispatch, I can configure the Thank You form if it is enabled.
- As an anonymous user, I can submit a Thank You form submission for someone with a UI email address.
- As a supervisor of someone receiving a Thank You form submission, I receive a copy of the email if the form is configured to do so.
- As the maintainer of an email address configured to receive a copy of a Thank You form submission, I receive said copy.

### Test existing Dispatch config is unchanged after database update
- `blt sync --site sitenow.uiowa.edu`
- Make sure api_key is still set in `sitenow_dispatch.settings`
